pkg/controller/history/controller_history.go

```diff
@@ -125,15 +126,19 @@ func EqualRevision(lhs *apps.ControllerRevision, rhs *apps.ControllerRevision) b
 		return lhs == rhs
 	}
 	if hs, found := lhs.Labels[ControllerRevisionHashLabel]; found {
-		hash, err := strconv.ParseInt(hs, 10, 32)
-		if err == nil {
+		hash, err := strconv.ParseUint(hs, 10, 32)
+		if err != nil {
+			glog.Errorf("Error parsing hash %s as uint32 for controller revision %s: %v", hs, lhs.Name, err)
```

Comment:
We usually prefer `utilruntime.HandleError()` instead of directly logging, in the case of errors that we ignore.

However, if I'm right about the intended semantics of this function, I don't think we should log/handle this error at all. If it's true that looking at the hash is an optional optimization, we shouldn't spam the logs just because we're falling back to the deep-compare.